### PR TITLE
It2 it3 bs

### PR DESCRIPTION
**Pull request by brennacodes**
- I had to update the `SalesEngine` class to match the interaction pattern an to make it work for `invoices` class, as well as additional classes that will be added later on. This update forced me to make minor adjustments to the other `repo` classes as well. 
- Everything was re-tested which is why there are minor changes to the test format. I wanted to make sure everything matched for clarity (and to make sure it all passed as expected). 
- Every one of the `repo` tests are currently passing, and the `SalesEngine` is flexible enough to take completely different hash arguments depending on what class is being instantiated. 
- In addition, `@all` no longer holds the actual data, it just points to it in the `.all` method.

### CODE REVIEW

**Comment by Rileybmcc** [comment]
Good job!

**Comment by alepbloyd** [comment]
Looking through this now! Will let you know if I have any questions on it.

File: test/merchant_test.rb [hunk 1]
```diff
@@ -1,34 +1,27 @@
 require 'simplecov'
 SimpleCov.start
-require './lib/merchant'
-require 'rspec'
-
-RSpec.describe Merchant do
+require './lib/helper'
+
+RSpec.describe InvoiceRepository do
```
Comment (alepbloyd) [inline test/merchant_test.rb]: Should be `RSpec.describe Merchant do` right?

File: test/merchant_test.rb [hunk 2]
```diff
@@ -1,34 +1,27 @@
...
+  }))}
 
   it 'exists' do
-    merchant = Merchant.new({
-      :id => 5,
-      :name => "Turing School"
-      })
-
-    expect(merchant).to be_instance_of(Merchant)
+    expect(merchant_repository).to be_instance_of(Merchant)
```
Comment (alepbloyd) [inline test/merchant_test.rb]: This is currently returning the MerchantRepository so is failing, but could be changed to `expect(merchant_repository.first).to be_instance_of(Merchant)` and it should pass I think

File: test/merchant_test.rb [hunk 3]
```diff
@@ -1,34 +1,27 @@
...
   end
 
   it 'returns id' do
```
Comment (alepbloyd) [inline test/merchant_test.rb:16]: Getting 'undefined method 'id' for #<Array>:blahblahblah'

image

File: test/merchant_test.rb [hunk 4]
```diff
@@ -1,34 +1,27 @@
...
-    merchant = Merchant.new({
-      :id => 5,
-      :name => "Turing School"
-      })
-
     expect(merchant.id).to eq(5)
   end
 
   it 'returns name' do
```
Comment (alepbloyd) [inline test/merchant_test.rb:20]: Same as above! 'merchant' seems to be an array, so it's not letting us call .name on it. 

image

**Review by alepbloyd — CHANGES_REQUESTED** [review]
@brennacodes This is huge refactoring and you are a champion thank you thank you. 

I added a handful of comments about little things to fix (I would go ahead and do them, but I worry that since this is my first time looking into the refactoring, I'll break more than I help with trying to fix them), but everything seems pretty minor but worth fixing before we merge. Lemme know if you want my help with any of it!

File: test/merchant_test.rb (hunk 1 above)
Comment (brennacodes) [reply test/merchant_test.rb]: fixed

File: test/merchant_test.rb (hunk 2 above)
Comment (brennacodes) [reply test/merchant_test.rb]: fixed

File: test/merchant_test.rb (hunk 3 above)
Comment (brennacodes) [reply test/merchant_test.rb:16]: fixed

File: test/merchant_test.rb (hunk 4 above)
Comment (brennacodes) [reply test/merchant_test.rb:20]: fixed

**Review by alepbloyd — APPROVED** [review]

**Pull request merged**